hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/MemberStateImpl.java

```diff
@@ -417,10 +418,10 @@ public JsonObject toJson() {
         return root;
     }
 
-    private static void serializeMap(JsonObject root, String key, Map<String, ? extends JsonSerializable> map) {
+    private static void serializeMap(JsonObject root, String key, Map<String, ?> map) {
         final JsonObject jsonObject = new JsonObject();
-        for (Entry<String, ? extends JsonSerializable> e : map.entrySet()) {
-            jsonObject.add(e.getKey(), e.getValue().toJson());
+        for (Entry<String, ?> e : map.entrySet()) {
+            jsonObject.add(e.getKey(), ((JsonSerializable) e.getValue()).toJson());
```

Comment:
To be on the safe side, I think we should add a type check here, as we don't want one stat to prevent all others from being sent in case of a casting error.

Reply:
Addressed review comments, now all casts are done after a check. Feel free to take a look.